### PR TITLE
Lesson-1-8

### PR DESCRIPTION
**Pull request by leipei92**
Lesson-1-8
Fetching datâ to/from airtable

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -3,60 +3,108 @@ import TodoList from './TodoList';
 import AddTodoForm from './AddTodoForm';
 
 
-const App = () => {
-  const [todoList, setTodoList] = React.useState([]);
-  const [isLoading, setIsLoading] = React.useState(true);
 
+const AIRTABLE_API_URL = `https://api.airtable.com/v0/${process.env.REACT_APP_AIRTABLE_BASE_ID}/${process.env.REACT_APP_TABLE_NAME}`;
```
Comment (JoHeyJo) [inline src/App.js:7]: Great use of a global variable here!

File: src/App.js [hunk 2]
```diff
@@ -3,60 +3,108 @@ import TodoList from './TodoList';
...
+      setIsLoading(false)
+    } catch (error) {
+      console.log(error.message)
+    }
   }, [])
 
-  React.useEffect(() => {
-    if (!isLoading) {
-      localStorage.setItem('savedTodoList', JSON.stringify(todoList))
+  const addTodo = async ({ title }) => {
```
Comment (JoHeyJo) [inline src/App.js:38]: Great job implementing this functionality! Something to consider: this function is doing a lot. It's posting to the API and it's also updating your state. Consider separating those concerns and streamlining the functionality of this function.

File: src/App.js [hunk 3]
```diff
@@ -3,60 +3,108 @@ import TodoList from './TodoList';
...
     }
-  }, [todoList, isLoading]);
+  };
+
 
-  const removeTodo = (id) => {
-    setTodoList(prevTodoList => {
-      return prevTodoList.filter(
-        (todo) => todo.id !== id
+  const removeTodo = async (id) => {
```
Comment (JoHeyJo) [inline src/App.js:71]: This is fantastic! Love that you implemented delete functionality!

**Review by JoHeyJo — APPROVED** [review]
Great job implementing the logic to interact with an API. Small suggestion to help improve your code but I'm approving. Well done!

File: src/App.js [hunk 4]
```diff
@@ -3,60 +3,108 @@ import TodoList from './TodoList';
...
+    try {
+      const response = await fetch(`${AIRTABLE_API_URL}/${id}`, {
```
Comment (JoHeyJo) [inline src/App.js:73]: Depending on what the return object is from your delete call, you can potentially reuse your state setter function for your `delete()` and `addTodo()`

File: src/App.js (hunk 1 above)
Comment (leipei92) [reply src/App.js:7]: It was a suggestion from one of our mentors and I really liked it. I try to work on not repeating myself unnecessary but it's not always easy.

**Pull request merged**